### PR TITLE
refactor/duplicated-widget-methods

### PR DESCRIPTION
**Pull request by unai-reliefapp**
# Description

- refactor: organize editor, summary-card-item-content components and widget service imports alphabetically 
- refactor: move content click action handler for editor and summary card item content into widget service as it's shared across widgets
- refactor: GraphQLVariables methods across widgets inside the widget service

## Useful links

- Please insert link to ticket
- Please insert link to back-end branch if any
- Please insert any useful link ( documentation you used for example )

## Type of change

Please delete options that are not relevant.

- [X] Improvement (refactor or addition to existing functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

- [ ] Test A
- [ ] Test B

## Screenshots

Please include screenshots of this change. If this issue is only back-end related, and does not involve any visual change of the platform, you can skip this part.

# Checklist:

( * == Mandatory ) 

- [x] * I have set myself as assignee of the pull request
- [x] * My code follows the style guidelines of this project
- [x] * Linting does not generate new warnings
- [x] * I have performed a self-review of my own code
- [ ] * I have put the ticket for review, adding the oort-frontend team to the list of reviewers
- [x] * I have commented my code, particularly in hard-to-understand areas
- [x] * I have put JSDoc comment in all required places
- [x] * My changes generate no new warnings
- [ ] * I have included screenshots describing my changes if relevant
- [x] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# More explanation
https://www.loom.com/share/05a716d61b9744faaf51fb304c21d1e5?sid=f87cf896-582a-4f76-93ae-8ceed801b145

### CODE REVIEW

File: libs/shared/src/lib/services/widget/widget.service.ts [hunk 1]
```diff
@@ -13,18 +23,31 @@ import { ShadowDomService } from '@oort-front/ui';
...
   constructor(
     private restService: RestService,
     @Inject(DOCUMENT) private document: Document,
-    private shadowDomService: ShadowDomService
-  ) {}
+    private shadowDomService: ShadowDomService,
+    private contextService: ContextService,
+    @Optional()
+    @SkipSelf()
+    private dashboardAutomationService: DashboardAutomationService,
```
Comment (AntoineRelief) [inline libs/shared/src/lib/services/widget/widget.service.ts]: @unai-reliefapp 
just to inform you as I know you're working on a different project
I tried to run it with the back-office but the dashboard automation service is never defined, and automation rules won't work due to that

Comment (unai-reliefapp) [reply libs/shared/src/lib/services/widget/widget.service.ts]: Now that I see makes sense, the widget service instance is created on root, therefor the dashboard automation service instance does not exist yet when the widget service is created.

The thing would be to maybe create a public method in the widget service that receives the dashboard automation service instance from the component to assign it to a private dashboard automation property in the widget service

```
In component

constructor(
...
 @Optional()
    @SkipSelf()
    private dashboardAutomationService: DashboardAutomationService,
){
this.widgetService.initializeDashboardService(dashboardAutomationService);
{

In widget service

public initializeDashboardService(dashboardAutomationService){
this.dashboardAutomationService = dashboardAutomationService;
}
```
I think that should do it

KR, Unai

File: libs/shared/src/lib/services/widget/widget.service.ts [hunk 2]
```diff
@@ -67,4 +81,139 @@ export class WidgetService {
...
+   *
+   * @param event Click event received from the widget
+   * @param widgetNodeName Current clicked widget node name
+   * @param dashboardAutomationService Dashboard automation service, provided by widget
+   * @param automationRules Automation rules from the widget settings to trigger if needed
+   */
+  public handleWidgetContentClick(
+    event: PointerEvent,
+    widgetNodeName: string,
+    dashboardAutomationService: DashboardAutomationService,
```
Comment (AntoineRelief) [inline libs/shared/src/lib/services/widget/widget.service.ts:96]: @unai-reliefapp 
see resolution for the dashboard automation service issue
provided by the widget directly

File: libs/shared/src/lib/components/widgets/summary-card/summary-card.component.ts [hunk 3]
```diff
@@ -263,18 +264,13 @@ export class SummaryCardComponent
...
-      mapping = this.contextService.replaceFilter(mapping);
-      mapping = this.replaceWidgetVariables(mapping);
-      this.contextService.removeEmptyPlaceholders(mapping);
-      return mapping;
-    } catch {
-      return null;
-    }
+    return this.widgetService.mapGraphQLVariables(
+      this.settings.card?.referenceDataVariableMapping as any,
+      this.replaceWidgetVariables.bind(this)
```
Comment (AntoineRelief) [inline libs/shared/src/lib/components/widgets/summary-card/summary-card.component.ts:269]: @unai-reliefapp 
another issue I found, about the custom placeholders based on widget
there, search & sort
this should solve it, and it simplifies the code, as only one time the removeEmptyPlaceholders needs to be called

**Pull request merged**